src/valid/function.rs

```diff
@@ -565,6 +565,22 @@ impl super::Validator {
                     name: argument.name.clone().unwrap_or_default(),
                 });
             }
+
+            // Pointers to unsized values are permitted, but only if they're structs
+            // with an unsized final member, not runtime-sized arrays.
+            let argument_inner = &module.types[argument.ty].inner;
+            if let crate::TypeInner::Pointer { base, class: _ } = *argument_inner {
+                let base_info = &self.types[base.index()];
+                if !base_info.flags.contains(TypeFlags::SIZED) {
```

Comment:
would it be easier (and still correct) to simply ensure that `TypeFlags::DATA` is there for the base type?
if it's not a structure, the general type validation should see that and complain

Reply:
You're right, type validation should forbid `ptr<_, array<f32>>`. (It doesn't, but we can fix that.) In that light, the existing check that argument types are sized should handle everything.

Reply:
Actually, we must permit pointers to dynamically sized arrays, since otherwise we have no type for the intermediate `AccessIndex` expression that refers to a dynamically sized array appearing as the last member of a struct.

But making such pointers non-`DATA` (and actually requiring local variables to be `SIZED` `DATA`) prevents the cases we care about preventing.